### DISCUSSION TITLE
Mass loss in simulating multiphase flow

### DISCUSSION THREAD

**Discussion by alimostafavi24** [idaholab/moose#24414] [Q&A Modules: Phase field]
Hello,

I am using coupled Cahn Hilliard Navier Stokes equation to simulate the droplet impact problem. The result that I am getting right now seems to be physically correct.

<img width="852" alt="Screen Shot 2023-05-17 at 12 25 26 PM" src="https://github.com/idaholab/moose/assets/119982603/d72a0fea-ba81-43dc-9cff-bccd108c4d84">

<img width="780" alt="Screen Shot 2023-05-17 at 12 25 36 PM" src="https://github.com/idaholab/moose/assets/119982603/312d2532-dde7-418f-aee0-139621f3cd4c">

Unfortunately I am witnessing a continuous mass loss. Is it because of the advection term that is added to the Cahn Hilliard equation? If this is the case, what are my options to correct this? (I think this is very similar to the rotating circle problem in level set module)

![5](https://github.com/idaholab/moose/assets/119982603/a52646a1-5b15-4818-a15e-ed3f8a6a3b93)

Thanks,

**Comment by GiudGiud** [comment]
Hello

If you remove the CH equations, or just the advection term in the CH equation, do you still observe this mass loss?

Guillaume

**Reply by alimostafavi24** [reply]
I removed the advection term in the CH equation. The area is still decreasing, but the rate of the decrease is much smaller.

**Reply by hugary1995** [reply]
Can you elaborate how you compute the total mass?

**Reply by alimostafavi24** [reply]
```
  [./area]
    type = LevelSetVolume
    threshold = 0.5
    variable = c
    location = outside
    execute_on = 'initial timestep_end'
  [../]
```
Since this is a 2d axisymmetric model, it is actually the volume.

Thanks,
Ali

**Reply by hugary1995** [reply]
Does ElementIntegralVariable report the same behavior?

**Reply by alimostafavi24** [reply]
Yes, I have both of them in the postprocessor.  ElementIntegralVariablePostprocessor also reports some continuous decrease.
```

[Postprocessors]


  # for testing we also monitor the total solute amount, which should be conserved.
  [./total_solute]
    type = ElementIntegralVariablePostprocessor
    variable = c
  [../]

  [./area]
    type = LevelSetVolume
    threshold = 0.5
    variable = c
    location = outside
    execute_on = 'initial timestep_end'
  [../]

[]

```

**Reply by lindsayad** [reply]
@dschwen have you done CH with advection? @alimostafavi24 can you share your phase-field/navier-stokes input file?

**Comment by ykvishal** [comment]
Did you find reasons for your mass loss issue? I have another question, which may not be useful. Are you applying mesh adaptivity (and/or time stepper)? If yes, can you share mesh, adaptivity and time-stepper block?

**Reply by alimostafavi24** [reply]
I think there are 2 reasons for that.
1. The advection term
2. The order parameter is not bounded which leads to mass loss.


Yes.

```
    [./TimeStepper]
    type = ConstantDT
    dt = 1e-6
  [../]

[Adaptivity]
  initial_steps = 5
  steps = 1
  marker = error
  [Indicators]
    [jump]
      type = GradientJumpIndicator
      variable = c
    []
  []
  [Markers]
    [error]
      type = ErrorFractionMarker
      coarsen = 0.95
      refine = 0.9
      indicator = jump
    []
  []
  max_h_level = 5
[]
```

**Reply by ykvishal** [reply]
You could run one identical simulation, but without mesh adaptivity and check mass loss again. Sometimes CH equation does not work well with mesh adaptivity.

**Reply by alimostafavi24** [reply]
You might be right!
![massloss](https://github.com/idaholab/moose/assets/119982603/0342b746-d95d-4046-ba40-a45bb28010df)

**Reply by ykvishal** [reply]
This looks interesting. Can you share mesh block for both cases ( with adapvity if any)?

**Comment by alimostafavi24** [comment]
<img width="819" alt="Screen Shot 2023-05-25 at 5 41 30 PM" src="https://github.com/idaholab/moose/assets/119982603/27f234fd-bf7e-4e1c-ae2d-4e79f2b41f02">

**Reply by ykvishal** [reply]
Thanks, I wanted to see mesh block from your input files. Can you double nx and ny in your right hand side simulation while keeping everything same? I want to see if there is any improvement.
